### PR TITLE
Update package after v0.3.0 release

### PR DESCRIPTION
**Pull request by rasolca**
Added modification from spack repo as well.

### CODE REVIEW

**Review by albestro — APPROVED** [review]
LGTM

File: spack/packages/dla-future/package.py [hunk 1]
```diff
@@ -134,7 +140,7 @@ def cmake_args(self):
         args.append(self.define_from_variant("BUILD_SHARED_LIBS", "shared"))
 
         # BLAS/LAPACK
-        if "^mkl" in spec:
+        if self.spec["lapack"].name in INTEL_MATH_LIBRARIES:
```
Comment (RMeli) [inline spack/packages/dla-future/package.py:143]: For reference, related to https://github.com/spack/spack/pull/41003

**Review by RMeli — APPROVED** [review]

**Review by msimberg — APPROVED** [review]

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Pull request merged**